cmd/dra-example-kubeletplugin/state.go

```diff
@@ -173,3 +193,67 @@ func (s *DeviceState) prepareDevices(claim *resourceapi.ResourceClaim) (Prepared
 func (s *DeviceState) unprepareDevices(claimUID string, devices PreparedDevices) error {
 	return nil
 }
+
+func (pds PreparedDevices) GetDevices() []*drapbv1.Device {
+	var devices []*drapbv1.Device
+	for _, pd := range pds {
+		devices = append(devices, &pd.Device)
+	}
+	return devices
+}
+
+type PtrHasGroupVersionKind[T any] interface {
+	*T
+	GroupVersionKind() schema.GroupVersionKind
+}
+
+// GetMergedOpaqueDeviceConfig merges configs across device classes and resource claims for a given device request.
+func GetMergedOpaqueDeviceConfig[T any, TI PtrHasGroupVersionKind[T]](request string, claim *resourceapi.ResourceClaim, classes map[string]*resourceapi.DeviceClass) (T, error) {
+	// Collect all configs in reverse priority order. Device class configs have
+	// lower priority than resource claim configs. Within either a device class
+	// or a resource claim, configs earlier in the list have lower priority
+	// than configs later in the list.
+	var allConfigs []resourceapi.DeviceConfiguration
+	for _, config := range classes[request].Spec.Config {
+		allConfigs = append(allConfigs, config.DeviceConfiguration)
+	}
+	for _, config := range claim.Spec.Devices.Config {
+		if !slices.Contains(config.Requests, request) {
+			continue
+		}
+		allConfigs = append(allConfigs, config.DeviceConfiguration)
+	}
+
+	// Merge all configs from the allConfigs slice in the order they were
```

Comment:
It might be useful for testing to factor this out into a function.

Reply:
I haven't done this yet, but if you feel strongly I can.

Reply:
It's not critical to this PR.